Modules/CTP/src/RawDataReaderCheck.cxx

```diff
@@ -0,0 +1,147 @@
+  ILOG(Debug, Devel) << "Run " << getActivity()->mId << ", type: " << getActivity()->mType << ", beam: " << getActivity()->mBeamType << ENDM;
+  // and you can get your custom parameters:
+  ILOG(Debug, Devel) << "custom param physics.pp.myOwnKey1 : " << mCustomParameters.atOrDefaultValue("myOwnKey1", "physics", "pp", "default_value") << ENDM;
+
+  int runNumber = getActivity()->mId;
+
+  o2::ccdb::CcdbApi api;
+  api.init("http://alice-ccdb.cern.ch");
+  auto& ccdbMgr = o2::ccdb::BasicCCDBManager::instance();
+  auto soreor = ccdbMgr.getRunDuration(runNumber);
+  uint64_t timeStamp = (soreor.second - soreor.first) / 2 + soreor.first;
+  std::string srun = std::to_string(runNumber);
+  map<string, string> metadata; // can be empty
+
+  auto lhcifdata = ccdbMgr.getSpecific<o2::parameters::GRPLHCIFData>("GLO/Config/GRPLHCIF", timeStamp, metadata);
+  auto bfilling = lhcifdata->getBunchFilling();
+  std::vector<int> bcs = bfilling.getFilledBCs();
+  o2::ctp::BCMask *bcmask;
+  std::bitset<o2::constants::lhc::LHCMaxBunches> lhcBC_bitset = bcmask->BCmask;
+  lhcBC_bitset.reset();
+
+  for (auto const& bc : bcs) {
+    lhcBC_bitset.set(bc, 1);
+  }
+
+  int threshold = 10;
+
+  for (auto& [moName, mo] : *moMap) {
+
+    (void)moName;
+    if (mo->getName() == "histobc") {
+      auto* h = dynamic_cast<TH1F*>(mo->getObject());
+
+      result = Quality::Good;
+      result.addMetadata("BC_id", "good");
+      for (int i = 0; i < o2::constants::lhc::LHCMaxBunches; i++) {
+        if(lhcBC_bitset[i]&&h->GetBinContent(i+1) < threshold) {
+          result = Quality::Bad;
+          result.updateMetadata(Form("BC_id%d",i+1), "bad");
+          result.addReason(quality_control::FlagReasonFactory::Unknown(), "Bunch crossing is expected, but not measured");
+          break;
+        }
+        if(lhcBC_bitset[i]&&h->GetBinContent(i+1) == threshold ){
+          result = Quality::Medium;
+          result.updateMetadata(Form("BC_id%d",i+1), "medium");
+          result.addReason(quality_control::FlagReasonFactory::Unknown(), "Bunch crossing is expected, at the threshold of background");
+        }
+        if(!lhcBC_bitset[i]&&h->GetBinContent(i+1) == threshold){
+          result = Quality::Medium;
+          result.updateMetadata(Form("BC_id%d",i+1), "medium");
+          result.addReason(quality_control::FlagReasonFactory::Unknown(), "Bunch crossing not expected, at the threshold of background");
+        }
+        if(!lhcBC_bitset[i]&&h->GetBinContent(i+1) > threshold) {
+          result = Quality::Bad;
+          result.updateMetadata(Form("BC_id%d",i+1), "bad");
+          result.addReason(quality_control::FlagReasonFactory::Unknown(), "Bunch crossing not expected, but measured");
+          break;
+        }
+      }
+    }
+  }
+  cout<< "Result check" << endl;
+  if (result == Quality::Good) cout << "green " <<endl;
+  if (result == Quality::Bad) cout << "red " <<endl;
+  if (result == Quality::Medium) cout << "orange " <<endl;
```

Comment:
Please avoid using `cout`. If the log should go to InfoLogger while running at P2, please use the `ILOG` facility. It will also appear in the logs when running on a laptop, assuming the log level and severity is higher than the threshold value.

If you do not need these log, please remove them.